src/wallets/injected/MathWallet.ts

```diff
@@ -0,0 +1,164 @@
+import isMobile from "is-mobile";
+
+import { mathWalletIcon } from "../icons";
+import { InjectedWallet, WalletModule } from "../Wallet";
+import { transformActions } from "../actions";
+import { InjectedMathWallet } from "./InjectedMathWallet";
+import { transactions, utils } from "near-api-js";
+
+declare global {
+  interface Window {
+    nearWalletApi: InjectedMathWallet | undefined;
+  }
+}
+
+function setupMathWallet(): WalletModule<InjectedWallet> {
+  return function MathWallet({
+    options,
+    provider,
+    emitter,
+    logger,
+    updateState,
+  }) {
+    let wallet: InjectedMathWallet;
+
+    const isInstalled = () => {
+      return !!window.nearWalletApi;
+    };
+
+    const timeout = (ms: number) => {
+      return new Promise((resolve) => setTimeout(resolve, ms));
+    };
+
+    return {
+      id: "math-wallet",
+      type: "injected",
+      name: "Math Wallet",
+      description: null,
+      iconUrl: mathWalletIcon,
+
+      isAvailable() {
+        if (!isInstalled()) {
+          return false;
+        }
+
+        if (isMobile()) {
+          return false;
+        }
+
+        return true;
+      },
+
+      async init() {
+        await timeout(200);
+
+        if (!isInstalled()) {
+          throw new Error("Wallet not installed");
+        }
+
+        wallet = window.nearWalletApi!;
+
+        if (wallet.signer.account) {
+          await wallet.login({});
+        }
+      },
+
+      async signIn() {
+        if (!isInstalled()) {
+          //TODO update state
+        }
+
+        if (!wallet) {
+          await this.init();
+        }
+
+        const account = await wallet.login({});
+
+        if (!account) {
+          throw new Error("Failed to sign in");
+        }
+
+        logger.log(options.networkId);
+
+        updateState((prevState) => ({
+          ...prevState,
+          showModal: false,
+          selectedWalletId: this.id,
+        }));
+        emitter.emit("signIn");
+      },
+
+      async isSignedIn() {
+        return Boolean(wallet.signer.account);
+      },
+
+      async signOut() {
+        const res = await wallet.logout();
```

Comment:
Is is the same that this returns a `boolean`?

Reply:
I think it is a promise that returns a boolean value in the end:

Just to test it if we type in console: 

`nearWalletApi.logout()`;

We get the below data: 
```ts
Promise {<pending>}
[[Prototype]]: Promise
[[PromiseState]]: "fulfilled"
[[PromiseResult]]: true
```

Reply:
Yeah looks like it 👍